app/src/main/kotlin/com/simplemobiletools/calculator/activities/MoneyActivity.kt

```diff
@@ -42,22 +55,134 @@ class MoneyActivity : SimpleActivity(), Calculator , TaxCalculator {
         updateViewColors(money_holder, config.textColor)
         updateButtonColor(config.customPrimaryColor)
 
+        locationManager = getSystemService(Context.LOCATION_SERVICE) as LocationManager
+
+        //Checks for permission granted for location service
+        if (ActivityCompat.checkSelfPermission(this, android.Manifest.permission.ACCESS_FINE_LOCATION)!= PackageManager.PERMISSION_GRANTED) {
+            //if not granted then request permission
+            ActivityCompat.requestPermissions(this, arrayOf(android.Manifest.permission.ACCESS_FINE_LOCATION), GPS_REQUEST_CODE);
+        }
+        try {
+            locationManager!!.requestLocationUpdates(LocationManager.GPS_PROVIDER, MINIMUM_TIME_BETWEEN_UPDATES, MINIMUM_DISTANCE_CHANGE_FOR_UPDATES, this)
+        } catch (e: SecurityException) {
+            print("NO PERMISSION GRANTED!")
+        }
+
         getButtonIds().forEach {
             it.setOnClickListener { calc.numpadClicked(it.id); checkHaptic(it) }
         }
 
         btn_currency.setOnClickListener{ true } // TODO : Implement feature and connect
         btn_delete.setOnClickListener { calc.handleDelete(); checkHaptic(it) }
         btn_delete.setOnLongClickListener { calc.handleClear(); true }
-        btn_taxes.setOnClickListener{ calc.calculateTax() } // TODO : Implement feature and connect
         btn_tip.setOnClickListener { true } // TODO : Implement feature and connect
         result.setOnLongClickListener { copyToClipboard(result.value); true }
+        btn_taxes.setOnClickListener({ view -> taxLocationStrat() })
 
         AutofitHelper.create(result)
 
     }
 
-     override fun spawnTaxModal() {
+    override fun onLocationChanged(location: Location) {
+        currentLatitude = location.latitude
+        currentLongitude = location.longitude
+        val message = String.format("New Location \n Longitude: %1\$s \n Latitude: %2\$s",
+                location.longitude, location.latitude
+        )
+        Toast.makeText(applicationContext, message, Toast.LENGTH_LONG).show()
+    }
+
+    override fun onStatusChanged(s:String, i:Int, b:Bundle) {}
+
+    override fun onProviderDisabled(s:String) {
+        Toast.makeText(applicationContext,"Location service has been turned off", Toast.LENGTH_LONG).show()
+    }
+    override fun onProviderEnabled(s:String) {
+        Toast.makeText(applicationContext,"Locattion service has been turned on", Toast.LENGTH_LONG).show()
+    }
+
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        if(requestCode == GPS_REQUEST_CODE) {
+            if(grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                Log.e("PermissionTag","Permission Granted (if 0 then is granted): " + grantResults[0])
+            }
+        }
+    }
+
+    fun taxLocationStrat() {
+        var gps_enabled : Boolean = false;
```

Comment:
**Style catch:** Variables in Kotlin should use camelCase. The semicolon at the end of this line can be removed

Reply:
Good styling catch

Reply:
Great catch! This totally slipped me completely! What would I do without you guys!!!!! You are all the best!!!1!!